addons/rearm/functions/fnc_addVehicleMagazinesToSupply.sqf

```diff
@@ -0,0 +1,40 @@
+/*
+ * Author: GitHawk
+ * Adds all magazines of a vehicle to the supply.
+ *
+ * Arguments:
+ * 0: Ammo Truck <OBJECT>
+ * 1: Vehicle or Vehicle class <OBJECT/STRING>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [ammo_truck, tank] call ace_rearm_fnc_addVehicleMagazinesToSupply
+ * [ammo_truck, "B_MBT_01_arty_F"] call ace_rearm_fnc_addVehicleMagazinesToSupply
+ *
+ * Public: Yes
+ */
+#include "script_component.hpp"
+
+params [
+    ["_truck", objNull, [objNull]],
+    ["_vehicle", objNull, [objNull, ""]]
+];
```

Comment:
All that type checking is a bit redundant (not sure about performance difference), also lengthy.